create-release.py

```diff
@@ -124,7 +124,7 @@ def update_version_to_release() -> None:
         sed(_source("README.md"), r"elyra:dev ", f"elyra:{new_version} ")
         if config.rc is None and config.beta is None:
             # Update the stable version Binder link
-            sed(_source("README.md"), r"/v[0-9].[0-9].[0-9]?", f"/v{new_version}?")
+            sed(_source("README.md"), r"/v[0-9]\+\.[0-9]\+\.[0-9]\+", f"/v{new_version}?")
```

Comment:
I find that escaping the `+` doesn't work from the mac shell.  I also need `-r` instead of `-e` - so I'm not sure if those commands are sufficient "tests".

```bash
$ echo "v4.0.0" | sed -e "s#v[0-9]\+\.[0-9]\+\.[0-9]\+#v4.0.1#g"
v4.0.0
$ echo "v4.0.0" | sed -e "s#v[0-9]+\.[0-9]+\.[0-9]+#v4.0.1#g"
v4.0.0
$ echo "v4.0.0" | sed -r "s#v[0-9]+\.[0-9]+\.[0-9]+#v4.0.1#g"
v4.0.1
```

Because this particular replacement is manipulating a section of the URL, I think the trailing `'?'` needs to be present in the match element, otherwise, this replacement will result in `'/v{new_version}??'`.
```bash
$ echo "/v4.10.10?" | sed -r "s#/v[0-9]+\.[0-9]+\.[0-9]+#/v{new_version}?#g"
/v{new_version}??
```